### DISCUSSION TITLE
Roles are empty for users when logged in as super admin

### DISCUSSION THREAD

**Discussion by Jaydeep7248** [spatie/laravel-permission#2051] [General]
Hello,

When I logged in as super admin and check the other users role, it returns empty.

My super admin role is super-admin and team id is NULL, In model_has_roles, it is
role_id model_type model_id team_id
5          App\User    5              5

My super admin id is 5.

I have tried
For user id 1.
role_id model_type model_id team_id
1          App\User    1              1

```
$user = User::find(1);
$user->roles();
```

It returns empty.

I have tried
```
setPermissionsTeamId(1);
$user = User::find(1);
$user->roles();
```

but still it returns empty.

Can anyone please suggest me the correct way?

**Comment by erikn69** [comment]
Make a sql debug, try this also
```php
setPermissionsTeamId(1);
$user = User::find(1);
$user->roles()->get();
```
